python/ql/lib/semmle/python/dataflow/new/internal/DataFlowPrivate.qll

```diff
@@ -641,25 +641,32 @@ predicate jumpStepNotSharedWithTypeTracker(Node nodeFrom, Node nodeTo) {
 //--------
 // Field flow
 //--------
+/**
+ * Subset of `storeStep` that should be shared with type-tracking.
+ */
+predicate storeStepCommon(Node nodeFrom, ContentSet c, Node nodeTo) {
+  tupleStoreStep(nodeFrom, c, nodeTo)
+  or
+  dictStoreStep(nodeFrom, c, nodeTo)
+  or
+  moreDictStoreSteps(nodeFrom, c, nodeTo)
+  or
+  iterableUnpackingStoreStep(nodeFrom, c, nodeTo)
+}
```

Comment:
Why does this not contain `attributeStoreStep`?

Reply:
Also, can we characterise the steps that should go here? Are they the ones where `c` is precise content? (This question is partly about maintenance of this predicate.)

Reply:
Good point :+1: let me write a comment :+1:

About the attributeStoreStep, as you already found out, it's already covered in type-tracking with slightly different logic, so it would be a mistake to also include it here.

Reply:
Fixed :+1: